packages/legend-studio/src/stores/LegendStudioPlugin.ts

```diff
@@ -52,6 +52,13 @@ export type ClassPreviewRenderer = (
   _class: Class,
 ) => React.ReactNode | undefined;
 
+export type ModelLoaderExtensionConfiguration = {
```

Comment:
Suggestions to not have to do this ` this.editorStore.graphManagerState.graphManager as V1_PureGraphManager` and make it more pluginable

1. I would add an abstract class maybe `ModelGenerationConfiguration`, this can contain `key, label`.
2. Remove `key,label` from `ModelLoaderExtensionConfiguration` and add config: ModelGenerationConfiguration to it.
3. Add `generateModelFromConfiguration` to `AbstractPureGraphManager` which takes in `ModelGenerationConfiguration` and the graph as a param.
4. Implement `generateModelFromConfiguration` in V1_PureGraphManager similar to `generateModel` by providing something similar to `V1_ModelGenerator` to the plugin. Maybe `V1_ModelGeneratorFromConfiguration`. you can provide the function the config, v1_pmcd and v1_engine. 
5. in the mapping generation plugin, you would have MappingGenerationConfiguration (not in the v1 space but maybe in `src/models` folder) extend ModelGenerationConfiguration. this can hold the rest of your configs as well such as sourceMapping etc. 
This would allow you to leverage functions such as `pureModelContextDataToEntities` and have the plugin just be responsible for generating pmcd given a config, graph and v1_engine.

Reply:
thanks a lot! I followed your suggestions and created MappingGeneration_PureProtocolProcessorPlugin